mail/protocols/imap/src/main/java/com/fsck/k9/mail/store/imap/ImapFolder.java

```diff
@@ -640,6 +640,20 @@ public void fetch(List<ImapMessage> messages, FetchProfile fetchProfile,
             return;
         }
 
+        synchronized (this) {
```

Comment:
This feels wrong. It looks like `CommandSearch` is only missing a call to `ImapFolder.open()`.

Reply:
That would be nice, but in fact the connection is already closed at this point in remote search, even if I open it in `CommandSearch`. Actually, because of the different threads probably, the `close` in the `finally` block is immediately executed after the `open` :). In addition, the connection is also closed in `fetchMessage`. I don't know how to fix this properly (therefore: hotfix...)